### PR TITLE
Hook up controller with DB (and other good stuff)

### PR DESCRIPTION
**Pull request by justinkrol**
This PR is a bit large, but it fixes many things, and I don't necessarily want to split it up since it's all pretty interactive (unless you guys find it too hard to review).

Here's what it changes:
1. Actually create the database tables on setup (line 1 of application.properties)
2. Fix the process of adding students to a project (and actually saving this)
    - This was broken because of a previous todo where we left `@Transient` on the `restrictions` list for the project. Since we call `restrictions.contains(....)`, but `restrictions` wasn't being saved, we were getting a `NullPointerException`.
    - Fixed by using `@ElementCollection` (good find @DerekStride), and doing some setting (so that foreign keys get set)
3. Seed the database with a project and a couple of students (in Application.java)
4. Use the seeded data in the controller (in HomeController.java)
5. Save attributes that come from the `User` superclass
    - This is done by using `@MappedSuperclass` on `User` (thanks again to @DerekStride)

### CODE REVIEW

File: src/main/java/app/Application.java [hunk 1]
```diff
@@ -1,11 +1,63 @@
...
+            log.info("students found with findAll():");
+            log.info("-------------------------------");
+            for (Student student : studentRepository.findAll()) {
+                log.info(student.toString());
+            }
+            log.info("");
+
+            // fetch an individual student by ID
+            Project proj = projectRepository.findOne(1L);
+            log.info("hello.BuddyInfo found with findOne(1L):");
```
Comment (DerekStride) [inline src/main/java/app/Application.java]: 👋

Comment (noahsegal) [inline src/main/java/app/Application.java]: 👨‍👦‍👦

**Review by DerekStride — APPROVED** [review]
🦋 💸 👼 

:shipit:

**Comment by JFleming4** [comment]
I should 💭  to ✍️  out of the 🖥 before I leave.

👁 💓 🦋 - "erfly" + "s"

Hail Satan! 😈 ☄️ 🔥

**Pull request merged**